src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -578,9 +578,25 @@ def _pem_password_cb(self, password):
         userdata = _PasswordUserdata(password=password)
         return _pem_password_cb, userdata
 
+
+    def _get_mgf1_supported_padding_hashes(backend):
+        # Verified test Vectors for OAEP only cover SHA-1 and SHA-2 currently
+        # with OpenSSL >= 1.0.2.
+        if backend._lib.OpenSSL_version_num() >= 0x10002001:
+            _oaep_supported_padding_algorithms = (
+                hashes.SHA224, hashes.SHA256,
```

Comment:
Yeah I noticed this too, although it does pass the test with SHA-1 because of the way the logic is written but I don't think it should rely on that. However, it raised another question to me, we only have test vectors that cover the SHA-2 combinations not SHA-1 with SHA-2 in the same OAEP. Do we want to add vectors to test SHA-1+SHA-2 combinations or put in a restriction to mark this combination as unsupported?